app/controllers/org_admin/templates_controller.rb

```diff
@@ -250,26 +250,22 @@ def history
     def publish
       template = Template.find(params[:id])
       authorize template
-      # rubocop:disable Metrics/LineLength
-      if template.latest?
-        # Now make the current version published
-        if template.publish!
-          flash[:notice] = _("Your #{template_type(template)} has been published and is now available to users.")
-        else
-          flash[:alert] = _("Unable to publish your #{template_type(template)}.")
-        end
+      template.published = true
+      if template.save
+        # rubocop:disable LineLength
+        flash[:notice] = _("Your #{template_type(template)} has been published and is now available to users.")
+        # rubocop:enable LineLength
       else
-        flash[:alert] = _("You can not publish a historical version of this #{template_type(template)}.")
+        flash[:alert] = failure_message(template, _("publish"))
       end
-      # rubocop:enable Metrics/LineLength
       redirect_to request.referrer.present? ? request.referrer : org_admin_templates_path
     end
 
     # PATCH /org_admin/templates/:id/unpublish  (AJAX)
     def unpublish
       template = Template.find(params[:id])
       authorize template
-      versions = Template.where(family_id: template.family_id)
+      versions = Template.where(family_id: template.family_id, published: true)
```

Comment:
This feels like its better suited to life in the model too since we have `publish` methods there.